### PR TITLE
Utilize new hspec format

### LINKED ISSUE #6: Use Test.Hspec.Core.Format
`hspec-2.8.0` exposes `Test.Hspec.Core.Format`.  This is very similar to what @eborden suggested in https://github.com/hspec/hspec/issues/411 and might be a better fit to implement `hspec-junit-formatter`.

NB: Please also feel free to add links to `hspec-junit-formatter` and `hspec-expectations-json` to the Hspec docs.

### PR DESCRIPTION
**Pull request by eborden**
The new format module is more flexible than the previous `writeLn`
formatting system. This implements a new version of
the package that no longer needs all the indirection of printing,
parsing, and re-rendering a junit.xml file. Large swaths of code have
been deleted and additional improvements have been made.

## Removed
- Temp files are no longer necessary.
- Nested suites in the schema are no longer necessary for after parse normalization.
- `withConfig` is removed in favor of exposing the `Format` function directly.

## Improvements
- Test case duration is now supported via `itemDuration`.
- Failure locations are listed via `Location` provided by some results.
- `timestamp` in the resulting XML now displays the start time of formatting.

## Pending
- A golden tests has been added, but requires normalization of timestamp and times
  to run without error.

### CODE REVIEW

File: library/Test/HSpec/JUnit.hs [hunk 1]
```diff
@@ -1,115 +1,123 @@
 module Test.HSpec.JUnit
-  ( runJUnitSpec
-  , configWith
```
Comment (pbrisbin) [inline library/Test/HSpec/JUnit.hs]: Is it possible to keep these two functions, at their present types, defined in terms of `junitFormat`?

If so, we could make this addition as minor-only and still remove all the code and have a clean implementation. If no, oh well.

File: hspec-junit-formatter.cabal [hunk 2]
```diff
@@ -41,14 +40,28 @@ library
...
+    , containers
     , directory
     , exceptions
-    , hashable
-    , hspec
-    , hspec-core
-    , resourcet
-    , temporary
+    , filepath
+    , hspec-core <=2.8.1
```
Comment (pbrisbin) [inline hspec-junit-formatter.cabal]: We aren't doing upper bounds anymore. And don't you want a _lower_ bound now?

File: tests/Main.hs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+main = do
+  summary <- flip runSpec defaultConfig $ do
+    describe "XML output" $ do
+      it "matches golen file" $ do
+        -- pendingWith
+        --   "Need to normalize time and timestamp values to match golden"
+        makeJunitResults
+
+        golden <- readFile "tests/golden.xml"
+        test <- readFile ".temp/test.xml"
```
Comment (pbrisbin) [inline tests/Main.hs:23]: :+1: I'm into this sort of testing for this library.

**Review by pbrisbin — APPROVED** [review]
:clap:

File: hspec-junit-formatter.cabal (hunk 2 above)
Comment (eborden) [reply hspec-junit-formatter.cabal]: Lol, this was suppose to be a lower. It'll take some squinting for me to figure out if it is an upper or lower :laughing:

File: library/Test/HSpec/JUnit.hs (hunk 1 above)
Comment (eborden) [reply library/Test/HSpec/JUnit.hs]: Yeah, they could be kept.

File: tests/Main.hs (hunk 3 above)
Comment (mjgpy3) [reply tests/Main.hs:23]: Yeah, this is really cool

**Review by mjgpy3 — APPROVED** [review]
Looks good to me! To Pat's point it might be nicer to release a minor change if possible.

**Comment by eborden** [comment]
Looks like this will release automatically on merge. Any qualms?

File: .restyled.yaml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+exclude:
+  - "tests/golden.xml"
```
Comment (pbrisbin) [inline .restyled.yaml:2]: Note that you've overwritten (not appended to) the default global excludes. It's probably fine, but FYI.

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,13 @@
 
 None
 
+## [v1.0.1.0](https://github.com/freckle/hspec-junit-formatter/compare/v1.0.0.4...v1.0.1.0)
+
+- Format function can be used directly without `withConfig` or `runJUnitSpec`.
+- Test case duration is now supported.
+- Failure locations are listed for some result types.
+- Timestamps in the resulting XML now display the start time of formatting.
+
```
Comment (eborden) [inline CHANGELOG.md:11]: @pbrisbin @mjgpy3, I refreshed your review status so you can explicitly sign off.

**Review by pbrisbin — APPROVED** [review]
No qualms with auto-release. We usually merge code changes on their own, then do version bumps in isolated PRs; but I really see no problem with batching when it's a 1-PR release you're planning anyway.

File: .restyled.yaml (hunk 4 above)
Comment (eborden) [reply .restyled.yaml:2]: Is it possible to append?

**Review by mjgpy3 — APPROVED** [review]

File: .restyled.yaml (hunk 4 above)
Comment (pbrisbin) [reply .restyled.yaml:2]: Nope, feature requests welcome. If you want to append, the current suggestion is to re-specify the default then add to it:

```yaml
exclude:
  - "**/*.patch"
  - "**/node_modules/**/*"
  - "**/vendor/**/*"
  - ".github/workflows/**/*"
  - "tests/golden.xml"
```

https://github.com/restyled-io/restyled.io/wiki/Configuring-Restyled#exclude

**Pull request merged**